lib/active_model/serializer/collection_serializer.rb

```diff
@@ -44,8 +44,16 @@ def json_key
           explicit_serializer_class._type
         # 2. get from first serializer instance in collection
         key ||= (serializer = serializers.first) && serializer.json_key
-        # 3. get from collection name, if a named collection
-        key ||= object.respond_to?(:name) ? object.name && object.name.underscore : nil
+        if object.respond_to?(:name)
+          begin
+            # 3. use json_key from serializer from collection name
+            key ||= serializer_from_resource(object.name.safe_constantize.new, ActiveModel::Serializer, options).json_key
+          rescue NoSerializerError, NameError
```

Comment:
Sort of wishing I had merged https://github.com/rails-api/active_model_serializers/pull/1767 at this point...